core/src/map.cpp

```diff
@@ -582,69 +586,202 @@ void Map::captureSnapshot(unsigned int* _data) {
 
+void Map::updateCameraPosition(const CameraUpdate& _update, float _duration, EaseType _e) {
+
+    CameraPosition camera;
+    if ((_update.set & CameraUpdate::SET_CAMERA) != 0) {
+        camera = getCameraPosition();
+    }
+    if ((_update.set & CameraUpdate::SET_BOUNDS) != 0) {
+        camera = getEnclosingCameraPosition(_update.bounds[0], _update.bounds[1], _update.boundsPadding);
+    }
+    if ((_update.set & CameraUpdate::SET_LNGLAT) != 0) {
+        camera.longitude = _update.lngLat.longitude;
+        camera.latitude = _update.lngLat.latitude;
+    }
+    if ((_update.set & CameraUpdate::SET_ZOOM) != 0) {
+        camera.zoom = _update.zoom;
+    }
+    if ((_update.set & CameraUpdate::SET_ROTATION) != 0) {
+        camera.rotation = _update.rotation;
+    }
+    if ((_update.set & CameraUpdate::SET_TILT) != 0) {
+        camera.tilt = _update.tilt;
+    }
+    if ((_update.set & CameraUpdate::SET_ZOOM_BY) != 0) {
+        camera.zoom += _update.zoomBy;
+    }
+    if ((_update.set & CameraUpdate::SET_ROTATION_BY) != 0) {
+        camera.rotation += _update.rotationBy;
+    }
+    if ((_update.set & CameraUpdate::SET_TILT_BY) != 0) {
+        camera.tilt += _update.tiltBy;
+    }
+
+    if (_duration == 0.f) {
+        setCameraPosition(camera);
+    } else {
+        setCameraPositionEased(camera, _duration, _e);
+    }
+}
+
+void Map::setPosition(double _lon, double _lat) {
+    cancelCameraAnimation();
+
+    glm::dvec2 meters = impl->view.getMapProjection().LonLatToMeters({ _lon, _lat});
+    impl->view.setPosition(meters.x, meters.y);
+    impl->inputHandler.cancelFling();
+    impl->platform->requestRender();
+}
+
+void Map::getPosition(double& _lon, double& _lat) {
     glm::dvec2 meters(impl->view.getPosition().x, impl->view.getPosition().y);
     glm::dvec2 degrees = impl->view.getMapProjection().MetersToLonLat(meters);
     _lon = LngLat::wrapLongitude(degrees.x);
     _lat = degrees.y;
+}
+
+void Map::setZoom(float _z) {
+    cancelCameraAnimation();
+
+    impl->view.setZoom(_z);
+    impl->inputHandler.cancelFling();
+    impl->platform->requestRender();
+}
 
+float Map::getZoom() {
+    return impl->view.getZoom();
 }
 
-void Map::Impl::setZoomNow(float _z) {
+void Map::setRotation(float _radians) {
+    cancelCameraAnimation();
 
-    view.setZoom(_z);
-    inputHandler.cancelFling();
-    platform->requestRender();
+    impl->view.setRoll(_radians);
+    impl->platform->requestRender();
+}
 
+float Map::getRotation() {
+    return impl->view.getRoll();
 }
 
-void Map::setZoom(float _z) {
+void Map::setTilt(float _radians) {
+    cancelCameraAnimation();
 
-    impl->setZoomNow(_z);
-    impl->clearEase(EaseField::zoom);
+    impl->view.setPitch(_radians);
+    impl->platform->requestRender();
+}
 
+float Map::getTilt() {
+    return impl->view.getPitch();
 }
 
-void Map::setZoomEased(float _z, float _duration, EaseType _e) {
+CameraPosition Map::getEnclosingCameraPosition(LngLat _a, LngLat _b, int _buffer) {
```

Comment:
Seems like it should be pretty simple to allow separate buffer values for each dimension - that would match the behavior of other SDKs.